tests/test_cli.py

```diff
@@ -34,31 +45,105 @@ def test_init(runner):
 
 def test_track(runner):
     query_id = '49741'
-    query_name = 'poc'
-    filename = query_name + '.sql'
+    file_name = 'poc.sql'
 
     with runner.isolated_filesystem():
         with HTTMock(response_content):
             result = runner.invoke(cli.track, [
                 query_id,
-                query_name,
+                file_name,
                 '--redash_api_key',
                 'TOTALLY_FAKE_KEY'
             ])
 
         print_debug_for_runner(result)
 
         # Verify we save the query to the right file
-        assert os.path.isfile(filename)
-        with open(filename, 'r') as query:
+        assert os.path.isfile(file_name)
+        with open(file_name, 'r') as query:
             assert "SELECT" in query.read()
 
         # Verify we save the query_id to the config file
         assert os.path.isfile(conf_path)
         with open(conf_path, 'r') as conf_file:
             config = json.loads(conf_file.read())
-            assert query_name in config
-            assert config[query_name]['redash_id'] == query_id
+            assert file_name in config
+            assert config[file_name]['id'] == query_id
+
+def test_push_tracked(runner):
+    query_id = '49741'
+    file_name = 'poc.sql'
+
+    with runner.isolated_filesystem():
+        with HTTMock(response_content):
+            result = runner.invoke(cli.track, [
+                query_id,
+                file_name
+            ])
+
+        # Read the saved query
+        assert os.path.isfile(file_name)
+        with open(file_name, 'r') as fin:
+            query_before = fin.read()
+
+        query_after = query_before + "--appended"
+        # Overwrite with a new query
+        with open(file_name, 'w') as fout:
+            fout.write(query_after)
+
+        # Now push the result
+        with HTTMock(push_response):
+            push_result = runner.invoke(cli.push, [
+                file_name
+            ])
+
+        m = md5.new()
+        m.update(query_after)
+        expected_output = "Query ID {} updated with content from {} (md5 {})".format(
+            query_id, file_name, m.hexdigest())
+        assert push_result.output.strip() == expected_output
+
+def test_push_fail(runner):
+    query_id = '49741'
+    file_name = 'poc.sql'
+
+    with runner.isolated_filesystem():
```

Comment:
Totally optional suggestion: Consider refactoring the duplicated code into a separate function.

Reply:
fixed